tests/unit/design_document_tests.py

```diff
@@ -626,5 +687,355 @@ def test_get_info(self):
+                '{"store": true}); }\n}'
+            )
+        err = cm.exception
+        self.assertEqual(
+            str(err),
+            'Search index search001 does not exist in this design doc'
+        )
+
+    def test_delete_a_search_index(self):
+        """
+        Test deleting a search index from the DesignDocument dictionary.
+        """
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        ddoc.add_search_index(
+            'search001',
+            'function (doc) {\n  index("default", doc._id); '
+            'if (doc._id) {index("name", doc.name, '
+            '{"store": true}); }\n}'
+        )
+        self.assertEqual(
+            ddoc.get('indexes')['search001'],
+            {'index': 'function (doc) {\n  index("default", doc._id); '
+                'if (doc._id) {index("name", doc.name, '
+                '{"store": true}); }\n}',
+             'analyzer': 'standard'}
+        )
+        ddoc.delete_index('search001')
+        self.assertEqual(ddoc.get('indexes'), {})
+
+    def test_fetch_search_index(self):
+        """
+        Ensure that the document fetch from the database returns the
+        DesignDocument format as expected when retrieving a design document
+        containing search indexes.
+        """
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        search_index = ('function (doc) {\n  index("default", doc._id); '
+                        'if (doc._id) {index("name", doc.name, '
+                        '{"store": true}); }\n} ')
+        ddoc.add_search_index('search001', search_index, 'simple')
+        ddoc.add_search_index('search002', search_index, 'simple')
+        ddoc.add_search_index('search003', search_index, 'simple')
+        ddoc.save()
+        ddoc_remote = DesignDocument(self.db, '_design/ddoc001')
+        self.assertNotEqual(ddoc_remote, ddoc)
+        ddoc_remote.fetch()
+        self.assertEqual(ddoc_remote, ddoc)
+        self.assertTrue(ddoc_remote['_rev'].startswith('1-'))
+        self.assertEqual(ddoc_remote, {
+            '_id': '_design/ddoc001',
+            '_rev': ddoc['_rev'],
+            'indexes': {
+                'search001': {'index': search_index, 'analyzer': 'simple'},
+                'search002': {'index': search_index, 'analyzer': 'simple'},
+                'search003': {'index': search_index, 'analyzer': 'simple'}
+            },
+            'views': {}
+        })
+
+    def test_fetch_no_search_index(self):
+        """
+        Ensure that the document fetched from the database returns the
+        DesignDocument format as expected when retrieving a design document
+        containing no search indexes.
+        """
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        ddoc.save()
+        ddoc_remote = DesignDocument(self.db, '_design/ddoc001')
+        ddoc_remote.fetch()
+        self.assertEqual(set(ddoc_remote.keys()),
+                         {'_id', '_rev', 'indexes', 'views'})
```

Comment:
It's a bit confusing why the `ddoc_remote.keys()` contains an `indexes` entry if the ddoc has no indexes. Is that the one we create in the locally cached design document that gets stripped on save if it is empty? If so perhaps a comment to that effect in this test might help someone unfamiliar with the code understand this oddity.